### PR TITLE
[GeneratorBundle] Kuma install command

### PR DESCRIPTION
**Pull request by cv65kr**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | no

Described in doc.

### CODE REVIEW

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Output\OutputInterface;
+use Sensio\Bundle\GeneratorBundle\Command\Validators;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Sensio\Bundle\GeneratorBundle\Command\Helper\QuestionHelper;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Debug\Exception\ContextErrorException;
+
```
Comment (sandergo90) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: Add class docblock.

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+class InstallCommand extends ContainerAwareCommand
```
Comment (sandergo90) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: For new commands, we don't extend the ContainerAwareCommand anymore. It should work with service injections.

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+{
+    private $commandSteps = 0;
+
+    protected function configure()
+    {
+        $this
+            ->setName('kuma:install')
+            ->setDescription('KunstmaanCMS installator.')
```
Comment (sandergo90) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: "installer"

**Comment by cv65kr** [comment]
@sandergo90, First of all, I wanted it to be consistent, without extra bash file. Secondly, we have less fields to fill, and also we can selected demosite install without bash file edit. If You think is not necessary, i will close this PR.

**Comment by sandergo90** [comment]
@cv65kr Alright, maybe it's a good thing then!

**Comment by cv65kr** [comment]
@sandergo90, Changed according to your guidelines.

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+     * @var ContainerInterface
+     */
+    private $containter;
+
+    /**
+     * @param ContainerInterface $container Container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
```
Comment (Numkil) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: Symfony 4 is deprecating using the container directly, in the bundles we also have several attempts going on to remove all places where we are injecting and using the database. I do not think its a good idea to add new code which uses the container now. Better just inject the dependencies you want.

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            ["No", "Yes"],
+            0
+        );
+        $question->setErrorMessage('Option "%s" is invalid.');
+        $demositeOption = $questionHelper->ask($input, $output, $question);
+
+        $input->setOption('demosite', $demositeOption);
+        $input->setOption('bundle-name',  strtr($namespace, ['\\Bundle\\' => '', '\\' => '']));
+
+        $dir = $input->getOption('dir') ?: dirname($this->container->getParameter('kernel.root_dir')) . '/src';
```
Comment (Numkil) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: You do not need to inject the container to get this parameter. you can inject the parameter alone in the services.yml.

**Comment by Devolicious** [comment]
Hi @cv65kr, 

If you can change the injection of the `container` to just the `%kernel.root_dir%` parameter, like @Numkil said, it would be much nicer and we can merge this PR. If you have some time it would be much appreciated. (the merge conflict also please)

Thx!

**Comment by sandergo90** [comment]
@cv65kr Do you have the time to finish this? Thanks!

File: src/Kunstmaan/GeneratorBundle/Resources/config/services.yml [hunk 6]
```diff
@@ -47,3 +47,7 @@ services:
       tags:
         - { name: console.command }
 
+    Kunstmaan\GeneratorBundle\Command\InstallCommand:
+        arguments: ['%kernel.root_dir%']
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Resources/config/services.yml]: Change this to `%kernel.project_dir%` as the root_dir parameter is deprecated in symfony 4.2

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+            ['No', 'Yes'],
+            0
+        );
+        $question->setErrorMessage('Option "%s" is invalid.');
+        $demositeOption = $questionHelper->ask($input, $output, $question);
+
+        $input->setOption('demosite', $demositeOption);
+        $input->setOption('bundle-name', strtr($namespace, ['\\Bundle\\' => '', '\\' => '']));
+
+        $dir = $input->getOption('dir') ?: dirname($this->rootDir) . '/src';
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: When the service parameter is changed from `root_dir` to `project_dir`, the dirname function should be removed

File: src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+     */
+    private $commandSteps = 0;
+
+    /** @var string */
+    private $rootDir;
+
+    /**
+     * @param string $rootDir
+     */
+    public function __construct(string $rootDir)
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Command/InstallCommand.php]: When the service parameter is changed from root_dir to project_dir, this parameter and property should be renamed

**Pull request merged**